pipeline/vars/common.groovy

```diff
@@ -148,33 +148,55 @@ def runTestSuite() {
     return executeTest(cmd, instanceName)
 }
 
-def sendEMail(def subjectPrefix,def test_results) {
-    /*
-        Send an email notification.
-    */
-    def body = readFile(file: "pipeline/vars/emailable-report.html")
-    body += "<body><u><h3>Test Summary</h3></u><br />"
-    body += "<p>Logs are available at ${env.BUILD_URL}</p><br />"
-    body += "<table><tr><th>Test Suite</th><th>Result</th>"
-
-    for (test in test_results) {
-        def res = "PASS"
-        if (test.value != 0) {
-            res = "FAIL"
+def fetchEmailBodyAndReceiver(def subjectPrefix, def test_results) {
+    def to_list = "ceph-qe-list@redhat.com"
+    def jobStatus = "stable"
+    def failureCount = 0
+    
+    def body = ""
+
+    if(subjectPrefix == "Tier-0"){
+        body += "<table><tr><th>Test Suite</th><th>Result</th>"
+        for (test in test_results) {
+            def res = "PASS"
+            if (test.value != 0) {
+                res = "FAIL"
+                failureCount += 1
+            } 
+
+            body += "<tr><td>${test.key}</td><td>${res}</td></tr>"
         }
+    }
+    else if(subjectPrefix == "Tier-1"){
```

Comment:
This would conflict when Tier-1 FG jobs attempt to send out consolidated emails after execution of the test suites. IMO, the following changes need to occur

- introduce a third argument to `sendEmail` to differentiate if it is a job or stage
- Have a method to generate Job results table
- Have a method to generate Stage results table
- If possible set the default value to stage,
- If stage, then process the results of test suites
- If job, then process the results of Jobs

Reply:
Done. Introduced a new isStage parameter, which is true by default and processed based on that.